src/init/resolvers.ts

```diff
@@ -17,30 +17,94 @@
  */
 import { Logger } from 'winston';
 import { StandaloneArNSResolver } from '../resolution/standalone-arns-resolver.js';
+import { OnDemandArNSResolver } from '../resolution/on-demand-arns-resolver.js';
 import { TrustedGatewayArNSResolver } from '../resolution/trusted-gateway-arns-resolver.js';
 import { NameResolver } from '../types.js';
+import { AoIORead } from '@ar.io/sdk';
+import { CompositeArNSResolver } from '../resolution/composite-arns-resolver.js';
 
 export const createArNSResolver = ({
   log,
   type,
-  url,
+  standaloneArnResolverUrl,
+  trustedGatewayUrl,
+  networkProcess,
 }: {
   log: Logger;
   type: string;
-  url: string;
+  standaloneArnResolverUrl?: string;
+  trustedGatewayUrl?: string;
+  networkProcess?: AoIORead;
 }): NameResolver => {
   log.info(`Using ${type} for arns name resolution`);
   switch (type) {
+    case 'on-demand': {
```

Comment:
Why not have something as `ON_DEMAND_RETRIEVAL_ORDER` instead of `type`? That way the order can be configured by an environment variable and default for whatever we want.

Reply:
good thought - i think with that in mind we could get ride of the use of `ARNS_RESOLVER_TYPE` and rely just on a priority order. Instead of `ON_DEMAND_RETRIEVAL_ORDER` I think the name `ARNS_RESOLVER_PRIORITY_ORDER` or something similar.

Reply:
updated in [2f113e1](https://github.com/ar-io/ar-io-node/pull/189/commits/2f113e14defe3b68104182287fb9d70eff30045e)